Src/FluentAssertions/Numeric/NullableNumericAssertions.cs

```diff
@@ -81,5 +83,35 @@ public AndConstraint<NullableNumericAssertions<T>> BeNull(string because = "", p
         {
             return NotHaveValue(because, becauseArgs);
         }
+
+        /// <summary>
+        /// Asserts that the <paramref name="predicate" /> is satisfied.
+        /// </summary>
+        /// <param name="predicate">
+        /// The predicate which must be satisfied
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])"/> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because"/>.
+        /// </param>
+        public AndConstraint<NullableNumericAssertions<T>> Match(Expression<Func<T?, bool>> predicate,
+            string because = "",
+            params object[] becauseArgs)
+        {
+            if (predicate == null)
```

Comment:
🙃 If you're branching out from master, I've added a `Guard.ThrowIfArgumentIsNull` helper.